diff --git a/arch/arm/boot/dts/at91-sam9x60_curiosity.dts b/arch/arm/boot/dts/at91-sam9x60_curiosity.dts
index 75e6727b5e3a..c10d06431521 100644
--- a/arch/arm/boot/dts/at91-sam9x60_curiosity.dts
+++ b/arch/arm/boot/dts/at91-sam9x60_curiosity.dts
@@ -433,7 +433,7 @@ AT91_PIOA 19 AT91_PERIPH_A (AT91_PINCTRL_PULL_UP | AT91_PINCTRL_DRIVE_STRENGTH_H
 		};
 		pinctrl_sdmmc0_cd: sdmmc0_cd {
 			atmel,pins =
-				<AT91_PIOA 25 AT91_PERIPH_GPIO AT91_PINCTRL_NONE>;
+				<AT91_PIOD 2 AT91_PERIPH_GPIO AT91_PINCTRL_NONE>;
 		};
 	};
 
@@ -473,7 +473,7 @@ &sdmmc0 {
 	bus-width = <4>;
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_sdmmc0_default &pinctrl_sdmmc0_cd>;
-	cd-gpios = <&pioA 25 GPIO_ACTIVE_LOW>;
+	cd-gpios = <&pioD 2 GPIO_ACTIVE_LOW>;
 	disable-wp;
 	status = "okay";
 };
